pkg/strace/strace_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package strace
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+// It's not really easy to write a full up general tester for this.
+// Even the simplest commands on Linux have dozens of system calls.
+// The Go assembler should in principle let us write a 3 line assembly
+// program that just does an exit system call:
+// MOVQ $exit, RARG
+// SYSCALL
+// But that's for someone else to do :-)
+
+func TestNoCommandFail(t *testing.T) {
+	Debug = t.Logf
+	c, err := NewTraceCommand("/etc/hosts", "hi")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := c.Run(); err == nil {
+		t.Fatalf("Got nil, want a non-nil error")
+	}
+}
+
+func TestBasicStrace(t *testing.T) {
+	Debug = t.Logf
+	c, err := NewTraceCommand("/bin/echo", "hi")
+	if err != nil {
+		t.Fatal(err)
+	}
+	or, ow, err := os.Pipe()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("%v %v", or, ow)
+	c.SetOutput(ow)
+
+	if err := c.Run(); err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("Run returned")
+
+	b, err := ioutil.ReadAll(or)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("%s", string(b))
```

Comment:
Can you add a check to make sure `b` contains a write syscall with the string "hi\n"?

Reply:
nice idea!

Reply:
but it won't work as I'm not dumping data yet. Later on.